vm-image-spec.yaml

```diff
@@ -302,7 +302,32 @@ files:
           - slot_name
         values: [restart_lsn]
         query: |
-          select slot_name, restart_lsn from pg_replication_slots where slot_type = 'logical';
+          select slot_name, (restart_lsn - '0/0')::FLOAT8 from pg_replication_slots where slot_type = 'logical';
+
+      - metric_name: retained_wal
+        type: gauge
+        help: 'Retained WAL in inactive replication slots'
+        key_labels:
+          - slot_name
+        values: [retained_wal]
+        query: |
+          SELECT slot_name, pg_wal_lsn_diff(pg_current_wal_lsn(), restart_lsn)::FLOAT8 AS retained_wal
+          FROM pg_replication_slots
+          WHERE active = false;
+
+      - metric_name: wal_is_lost
```

Comment:
Would it make sense to expose the other statuses too? I.e. just expose `wal_status` as it is?

Reply:
I did that originally, but you can't expose a string as a value, so we can't have `replication_slot` as a key and `wal_status` as a value. I figured `wal_status=lost` is the most important metric anyway because the other three statuses are recoverable implicitly, whereas with `lost` you must recreate the replication slot.

Reply:
Also you can imagine with this scheme, we could sum across all computes to see how many broken replication slots we have

Reply:
Ok, makes sense